### PR TITLE
Add some TableMapEvent helper methods

### PR DESCRIPTION
**Pull request by huangjunwen**
This pr adds some type-related helper methods for `TableMapEvent`. https://github.com/siddontang/go-mysql/issues/479

They should behave the same as `Table_map_log_event::print_columns` (`mysqlbinlog --print-table-metadata`).

And a `flavor` field is added to `BinlogParser` and passed to `TableMapEvent` to distinguish mysql/mariadb, since there are some differences slightly in checking types.

### CODE REVIEW

**Comment by GregoryIan** [comment]
LGTM

**Comment by GregoryIan** [comment]
@csuzhangxc PTAL, if there're not any problems, we can merge this PR ASAP

File: replication/row_event.go [hunk 1]
```diff
@@ -492,6 +603,217 @@ func (e *TableMapEvent) bytesSlice2StrSlice(src [][]byte) []string {
...
+}
+
+// Below realType and IsXXXColumn are base from:
+//   table_def::type in sql/rpl_utility.h
+//   Table_map_log_event::print_columns in mysql-8.0/sql/log_event.cc and mariadb-10.5/sql/log_event_client.cc
+
+func (e *TableMapEvent) realType(i int) byte {
+
+	typ := e.ColumnType[i]
+	meta := e.ColumnMeta[i]
```
Comment (csuzhangxc) [inline replication/row_event.go:744]: How about putting this line above L748?

File: replication/row_event.go [hunk 2]
```diff
@@ -492,6 +603,217 @@ func (e *TableMapEvent) bytesSlice2StrSlice(src [][]byte) []string {
...
+	switch e.realType(i) {
+	case MYSQL_TYPE_STRING,
+		MYSQL_TYPE_VAR_STRING,
+		MYSQL_TYPE_VARCHAR,
+		MYSQL_TYPE_BLOB:
+		return true
+
+	case MYSQL_TYPE_GEOMETRY:
+		if e.flavor == "mariadb" {
+			return true
```
Comment (csuzhangxc) [inline replication/row_event.go:790]: How about highlight this special behavior as the comments of this func (`IsCharacterColumn`)?

**Pull request merged**